OpenRA.Mods.Common/Traits/BotModules/BotModuleLogic/BaseBuilderQueueManager.cs

```diff
@@ -335,7 +346,7 @@ ActorInfo ChooseBuildingToBuild(ProductionQueue queue)
 				var buildingVariantInfo = actorInfo.TraitInfoOrDefault<PlaceBuildingVariantsInfo>();
 				var variants = buildingVariantInfo?.Actors ?? Array.Empty<string>();
 
-				var count = playerBuildings.Count(a => a.Info.Name == name || variants.Contains(a.Info.Name));
+				var count = playerBuildings.Count(a => a.Info.Name == name || variants.Contains(a.Info.Name)) + (baseBuilder.BuildingsBeingProduced != null ? (baseBuilder.BuildingsBeingProduced.TryGetValue(name, out var num) ? num : 0) : 0);
```

Comment:
We can remove the null check on `baseBuilder.BuildingsBeingProduced` if we go with my other suggestion of initializing it at declaration.

Feel free to add some line breaks so this isn't such a long line.

Reply:
Fixed by using the suggestion above.

All tested.